src/baf/knot_vector.cc

```diff
@@ -143,5 +151,9 @@ void baf::KnotVector::InsertKnot(const ParamCoord &param_coord) {
 
 void baf::KnotVector::RemoveKnot(const ParamCoord &param_coord) {
   KnotSpan knot_span = GetKnotSpan(param_coord);
-  knots_.erase(knots_.begin() + knot_span.get());
+  if (!IsLastKnot(param_coord)) {
+    knots_.erase(knots_.begin() + knot_span.get());
+  } else if (IsLastKnot(param_coord)) {
```

Comment:
You can use else without the second if case, as it's the opposite of the first if statement.